sonar-scanner-engine/src/main/java/org/sonar/scanner/scan/report/JSONReport.java

```diff
@@ -130,10 +135,15 @@ private void writeJsonIssues(JsonWriter json, Set<RuleKey> ruleKeys, Set<String>
     json.name("issues").beginArray();
     for (TrackedIssue issue : getIssues()) {
       if (issue.resolution() == null) {
+        InputComponent component = componentStore.getByKey(issue.componentKey());
+        String componentKey = getModule(component).definition().getKeyWithBranch();
```

Comment:
It seems this assignment should go in the `else` block of the next `if` statement